src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java

```diff
@@ -100,13 +100,59 @@ public DefaultSerializer(final ILogger logger) {
+            setChildAdditionalData(jsonBackedObject,rawObject);
             return jo;
         } else {
             logger.logDebug("Deserializing a non-IJsonBackedObject type " + clazz.getSimpleName());
             return jsonObject;
         }
     }
 
+    /**
+     * Recursively sets additional data for each child object
+     *
+     * @param serializedObject   the parent object whose children will be iterated to set additional data
+     * @param rawJson            the raw json
+     */
+    @SuppressWarnings("unchecked")
+    private void setChildAdditionalData(IJsonBackedObject serializedObject, JsonObject rawJson) {
+        // Use reflection to iterate through fields for eligible Graph children
+        for (java.lang.reflect.Field field : serializedObject.getClass().getFields()) {
+            try {
+                Object fieldObject = field.get(serializedObject);
+
+                // If the object is a HashMap, iterate through its children
+                if (fieldObject instanceof HashMap) {
+                    @SuppressWarnings("unchecked")
+                    HashMap<String, Object> serializableChildren = (HashMap<String, Object>) fieldObject;
+                    Iterator<Entry<String, Object>> it = serializableChildren.entrySet().iterator();
+
+                    while (it.hasNext()) {
+                        Map.Entry<String, Object> pair = (Map.Entry<String, Object>)it.next();
+                        Object child = pair.getValue();
+
+                        // If the item is a valid Graph object, set its additional data
+                        if (child instanceof IJsonBackedObject) {
+                            AdditionalDataManager childAdditionalDataManager = ((IJsonBackedObject) child).additionalDataManager();
+                            childAdditionalDataManager.setAdditionalData(rawJson.get(field.getName()).getAsJsonObject());
+                            setChildAdditionalData((IJsonBackedObject) child,rawJson.get(field.getName()).getAsJsonObject());
+                        }
+                    }
+                }
+                // If the object is a valid Graph object, set its additional data
+                else if (fieldObject != null && fieldObject instanceof IJsonBackedObject) {
+                    IJsonBackedObject serializedChild = (IJsonBackedObject) fieldObject;
+                    AdditionalDataManager childAdditionalDataManager = serializedChild.additionalDataManager();
+
+                    childAdditionalDataManager.setAdditionalData(rawJson.get(field.getName()).getAsJsonObject());
+                    setChildAdditionalData((IJsonBackedObject) fieldObject,rawJson.get(field.getName()).getAsJsonObject());
+                }
+            } catch (IllegalArgumentException | IllegalAccessException e) {
+                logger.logError("Unable to access child fields of " + serializedObject.getClass().getSimpleName(), e);
```

Comment:
What is the argument for not throwing? Can comments be added here to that effect?

Reply:
I will add the comments for more clarification.

Reply:
I have raised a separate PR for this - #114 .